src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -666,6 +685,10 @@ private void BufferHeaderCollection(HttpHeaders headers)
                     // The Connection, Upgrade and ProxyConnection headers are also not supported in HTTP/3.
                     if (knownHeader != KnownHeaders.Host && knownHeader != KnownHeaders.Connection && knownHeader != KnownHeaders.Upgrade && knownHeader != KnownHeaders.ProxyConnection)
                     {
+                        // The length of the encoded name may be shorter than the actual name.
```

Comment:
Is it theoretically possible that the encoded name is _longer_ than the raw one? Not if we only ever use static encoding, right?

Reply:
Normally the encoded length will be either 1 byte (static index from the spec) or just the whole literal with some HPack overhead. That is, it _may_ be longer, but that's okay since this is just an >= estimate.